app/controllers/orders/detail.js

```diff
@@ -54,6 +57,25 @@ export default GoodcityController.extend({
     }
   }),
 
+  cancellationReasons: Ember.computed(function() {
+    return this.store.peekAll("cancellation_reason");
+  }),
+
+  cancellationReasonId: Ember.computed(function() {
+    return this.get("cancellationReasons.firstObject.id");
+  }),
+
+  isOtherCancellationReason: Ember.computed("cancellationReasonId", function() {
+    const selectedCancelledReason = this.store.peekRecord(
+      "cancellation_reason",
+      this.get("cancellationReasonId")
+    );
+    return (
+      selectedCancelledReason.get("name") ===
+      this.get("i18n").t("order_cancellation_reason.other").string
```

Comment:
maybe better to use a constant for this instead of a translation string. Might fail in Chinese. And it'll certainly break if we ever want to change the UI text